src/orange/detail/BIHBuilder.cc

```diff
@@ -131,19 +133,31 @@ void BIHBuilder::construct_tree(VecIndices const& indices,
         node.parent = parent;
         node.axis = p.axis;
 
-        auto ax = to_int(p.axis);
+        // Return the current bounding box, clipped by the supplied halfspace
+        auto get_shrunk_bbox
+            = [&bbox, axis = p.axis](Bound bound, FastBBox::real_type pos) {
+                  auto out_box = bbox;
+                  out_box.shrink(bound, axis, pos);
+                  return out_box;
+              };
 
-        node.bounding_planes[Edge::left].position
-            = p.bboxes[Edge::left].upper()[ax];
-        node.bounding_planes[Edge::right].position
-            = p.bboxes[Edge::right].lower()[ax];
+        // Populate left/right bounding planes
+        auto left_pos = p.bboxes[Side::left].upper()[to_int(p.axis)];
+        node.edges[Side::left].bounding_plane_pos = left_pos;
+        node.edges[Side::left].bbox = get_shrunk_bbox(Bound::hi, left_pos);
+
+        auto right_pos = p.bboxes[Side::right].lower()[to_int(p.axis)];
+        node.edges[Side::right].bounding_plane_pos = right_pos;
+        node.edges[Side::right].bbox = get_shrunk_bbox(Bound::lo, right_pos);
 
         // Recursively construct the left and right branches
-        for (auto edge : range(Edge::size_))
+        for (auto edge : range(Side::size_))
```

Comment:
```suggestion
        for (auto side : range(Side::size_))
```
maybe? Since you might have `Edge& edge = node.edges[side]`?